perception/sandbox/polygrasp/scripts/run_grasp.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+"""
+Connects to robot, camera(s), gripper, and grasp server.
+Runs grasps generated from grasp server.
+"""
+
+import time
+import os
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import torch
+import hydra
+import omegaconf
+
+from polygrasp.segmentation_rpc import SegmentationClient
+from polygrasp.grasp_rpc import GraspClient
+from polygrasp.serdes import load_bw_img
+
+
+def save_rgbd_masked(rgbd, rgbd_masked):
+    num_cams = rgbd.shape[0]
+    f, axarr = plt.subplots(2, num_cams)
+
+    for i in range(num_cams):
+        axarr[0, i].imshow(rgbd[i, :, :, :3].astype(np.uint8))
+        axarr[1, i].imshow(rgbd_masked[i, :, :, :3].astype(np.uint8))
+
+    f.savefig("rgbd_masked.png")
+    plt.close(f)
+
+
+def save_obj_masked(obj_masked_rgbd, obj_i, obj_mask_size):
+    plt.imshow(obj_masked_rgbd[:, :, :3])
+    plt.title(f"Object {obj_i}, mask size {obj_mask_size}")
+    plt.savefig(f"object_{obj_i}_masked")
+    plt.close()
+
+
+@hydra.main(config_path="../conf", config_name="run_grasp")
+def main(cfg):
+    print(f"Config:\n{omegaconf.OmegaConf.to_yaml(cfg, resolve=True)}")
+    print(f"Current working directory: {os.getcwd()}")
+
+    print("Initialize robot & gripper")
+    robot = hydra.utils.instantiate(cfg.robot)
+    robot.gripper_open()
+    robot.go_home()
+
+    print("Initializing cameras")
+    cfg.cam.intrinsics_file = hydra.utils.to_absolute_path(cfg.cam.intrinsics_file)
+    cfg.cam.extrinsics_file = hydra.utils.to_absolute_path(cfg.cam.extrinsics_file)
+    cameras = hydra.utils.instantiate(cfg.cam)
+
+    print("Loading camera workspace masks")
+    masks_1 = np.array(
+        [load_bw_img(hydra.utils.to_absolute_path(x)) for x in cfg.masks_1], dtype=np.float64
+    )
+    masks_2 = np.array(
+        [load_bw_img(hydra.utils.to_absolute_path(x)) for x in cfg.masks_2], dtype=np.float64
+    )
+
+    print("Connect to grasp candidate selection and pointcloud processor")
+    segmentation_client = SegmentationClient()
+    grasp_client = GraspClient(view_json_path=hydra.utils.to_absolute_path(cfg.view_json_path))
+
+    root_working_dir = os.getcwd()
+    for outer_i in range(1, 10):
```

Comment:
Why start at 1?

Reply:
My kitchen utensils were on the other bin that day, haha. You're right.